client-core/src/transaction_builder/raw_transfer_transaction_builder.rs

```diff
@@ -1171,14 +1177,14 @@ mod raw_transfer_transaction_builder_tests {
     fn create_public_key_witness(
         private_key: PrivateKey,
         public_key: PublicKey,
-        tx_id: TxId,
+        tx: &Transaction,
     ) -> TxInWitness {
         let signing_addr = create_transfer_addr(public_key.clone());
 
         let signer =
             KeyPairSigner::new(private_key, public_key).expect("should create KeyPairSigner");
         signer
-            .schnorr_sign(tx_id, &signing_addr)
+            .schnorr_sign(tx, &signing_addr)
             .expect("should sign transaction id")
     }
```

Comment:
this function is used in transaction creation and it won't work with HW wallets, as it requires PrivateKey